### PR TITLE
`Exam mode`: Change test case default visibility to after results release date

### PR DESCRIPTION
**Pull request by florian-glombik**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [ ] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Description
* Adjusted the default test case visibility depending on whether the exercise is an exam or course exercise _(and corresponding translations)_
* Fixing issues introduced by #7242 _(console error when loading student exam as instructor, change submitted state displayed twice)_
* Fixed issue in supporting_script introduced in #7379

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Students

1. `instructor`: create a course programming exercise
2. `instructor`: navigate to grading of the exercise and see that the visibility of the test cases is `Always`
3. `instructor`: create an exam programming exercise
4. `instructor`: navigate to grading of the exercise and see that the visibility of the test cases is `After Release Date of Results`
5. `instructor`: make some changes and see that `reset` rests to the expected default state
6. `student`: participate in the course exercise and see that you get immediate feedback
7. `student`: participate in the exam and see that you do only get the info whether your solution compiled

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [student-exam-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Client-Tests/app/exam/manage/student-exams/student-exam-detail.component.ts.html) | 90.21% | ✅ ❌ |
| [programming-exercise-grading-tasks-table.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/grading/tasks/programming-exercise-grading-tasks-table.component.ts.html) | 91.3% | ✅ ❌ |
| [programming-exercise-task.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/grading/tasks/programming-exercise-task/programming-exercise-task.component.ts.html) | 92.68% | ✅ ❌ |
| [utils.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/util/utils.ts.html) | 98.21% | ✅ ❌ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [ProgrammingExercise.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/ProgrammingExercise.html) | 89% | ✅ ❌ |
| [ProgrammingExerciseFeedbackCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.programming/ProgrammingExerciseFeedbackCreationService.html) | 94% | ✅ ❌ |
| [ProgrammingExerciseTestCaseService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.programming/ProgrammingExerciseTestCaseService.html) | 89% | ✅ ❌ |
| [ProgrammingExerciseTestCaseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5824/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/ProgrammingExerciseTestCaseResource.html) | 100% | ✅ |

### Screenshots
TODO

### CODE REVIEW

File: supporting_scripts/course-setup-quickstart/create_course.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def create_course(session):
     response = session.post(url, data=body, headers=headers)
 
     if response.status_code == 201:
-        logging.success(f"Created course {course_name} with shortName {course_short_name} \n {response.json()}")
+        logging.info(f"Created course {course_name} with shortName {course_short_name} \n {response.json()}")
```
Comment (florian-glombik) [inline supporting_scripts/course-setup-quickstart/create_course.py:91]: `logging.success` does not exist

**Review by vinceclifford — DISMISSED** [review]
Works as expected, demonstrated during testing session

**Review by milljoniaer — DISMISSED** [review]
tested in testing session, works as expected

**Review by laurenzfb — DISMISSED** [review]
demonstratedin testing session. Worked as described

**Review by Jan-Thurner — APPROVED** [review]
got a demo in a TS and it worked as expected, the function of the "Reset"-button imo is not clear from the UI.

**Review by sarpsahinalp — APPROVED** [review]
Works as expected, demonstrated during testing session

**Review by lennart-keller — DISMISSED** [review]
Tested during testing session. Works well

**Comment by florian-glombik** [comment]
State of this PR:
* Implementation finished
* Merge conflicts with Angular17 update need to be resolved
* E2E tests need to be adjusted properly _(Adjusting the E2E tests has led to a failure (which seemed only to occur in the remote environment - one of the last commits trying to fix the issue introduced an actual mistake))_

**Pull request closed without merging**